autotest/pyscripts/test_gdal_utils_cli.py

```diff
@@ -0,0 +1,135 @@
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+'''
+Test that command line gdal-utils are in PATH and run. We only test script
+starts and returns expected defaults. We do not test if script functions
+properly.
+
+Tested:
+    gdalcompare <no args>
+Returns:
+    Usage: gdalcompare.py [-sds] <golden_file> <new_file>
+
+Not tested:
+    gdalcompare file1.tif file2.tif
+Returns:
+    <valid results>
+
+Resources used:
+ - https://github.com/rhcarvalho/pytest-subprocess-example
+ - https://stackoverflow.com/questions/30528099/unittesting-python-code-which-uses-subprocess-popen
+ - https://www.digitalocean.com/community/tutorials/how-to-use-subprocess-to-run-external-programs-in-python-3
+ - https://stackoverflow.com/questions/13493288/python-cli-program-unit-testing
+'''
+from pathlib import Path
+import sys
+import subprocess
+import pytest
+
+# Skip if gdal-utils is not known to pip (and therefore not registered in
+# python 'site-packages' and 'Scripts')
+installed = subprocess.run([sys.executable, '-m', 'pip', 'show', 'gdal-utils'])
+if not installed.returncode == 0:
+    pytest.skip("The 'gdal-utils' package is not installed.", allow_module_level=True)
+
+utils = ['gdal2tiles', 'gdal2xyz', 'gdal_calc', 'gdal_edit', 'gdal_fillnodata',
+         'gdal_merge', 'gdal_pansharpen', 'gdal_polygonize', 'gdal_proximity',
+         'gdal_retile', 'gdal_sieve', 'gdalattachpct', 'gdalcompare', 'gdalmove',
+         'ogrmerge', 'pct2rgb', 'rgb2pct']
+
+here = Path(__file__).parent.absolute()
+outputs_dir = Path.joinpath(here, "cli_outs")
+
+
+def get_utils_responses():
+    '''Return dict of "utility_name: [stdout_msg, stderr_msg]"
+
+    pct2rgb:
+        ''
+        'usage: pct2rgb [-h] [-of gdal_format] [-rgba] [-b band] ...\n'
+
+   The messages are what we expect the program name to report to console
+   when called with no parameters.
+
+    The expected results must be in pre-generated in files named:
+
+        cli_outs/scriptname.stdout  # normal
+        cli_outs/scriptname.stderr  # errors
+
+    These output files can be generated with:
+
+        gdalcompare 1>gdalcompare.stdout 2>gdalcompare.stderr
```

Comment:
I really don't like having all those .stderr / .stdout files. This is going to be an extra hurdle for contributors.
Looking at the output it seems that almost all have "usage:" / "Usage:" in the stdout/stderr. And for the few ones that don't (gdal_proximity, gdal_sieve, gdalmove, ogrmerge.py), it seems that checking that the name of the script and " <" or " [" is present should be enough. Or it might actually be faster to edit them so they output "usage:" as well.

Reply:
I agree. I was just hesitant to add something like that to this PR; I don't want to overstep my bounds. I'll do this now though.

Reply:
Done in fa37316.  It was a lot easier than I envisioned. (I'd noticed in my earlier scans that the usage message is coming from some place hidden to me for many of the scripts. I thought that meant extended edits would be needed. It turns out all of the hidden ones already start with _usage_ so I didn't have to touch them. Yay!)